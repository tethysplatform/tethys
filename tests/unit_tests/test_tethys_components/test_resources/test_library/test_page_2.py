def test(lib):
    lib.register("react-plotly@1.0.0", "p")
    chart_data = {
        "river_id": "Test 123",
        "series": [{"x": 1, "y": 100}, {"x": 2, "y": 200}],
    }
    return lib.html.div(style=lib.Style(width="100vw", height="calc(100vh - 57px)"))(
        lib.tethys.Map(
            lib.ol.layer.Image(title="GEOGLOWS Streamflow Service")(
                lib.ol.source.ImageArcGISRest(
                    url="https://livefeeds3.arcgis.com/arcgis/rest/services/GEOGLOWS/GlobalWaterModel_Medium/MapServer"
                )
            ),
        ),
        lib.tethys.Panel(
            show=True,
            anchor="right",
            extent="50vw",
            title="Forecast",
        )(
            lib.bs.Container(
                lib.bs.Row(
                    lib.bs.Col(
                        (
                            lib.html.h2(f"Streamflow @ {chart_data['river_id']}")
                            if chart_data
                            else None
                        ),
                        lib.rc.LineChart(
                            width=700, height=500, data=chart_data["series"]
                        )(
                            lib.rc.CartesianGrid(strokeDasharray="3 3"),
                            lib.rc.XAxis(label="Date"),
                            lib.rc.YAxis(
                                label=lib.Props(
                                    value="Streamflow", angle=-90, position="insideLeft"
                                )
                            ),
                            lib.rc.Tooltip(),
                            lib.rc.Line(type="monotone", data_key="y"),
                        ),
                    )
                ),
            ),
        ),
    )
