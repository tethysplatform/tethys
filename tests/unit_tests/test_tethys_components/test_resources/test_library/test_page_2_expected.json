{"tagName": "div", "attributes": {"style": {"width": "100vw", "height": "calc(100vh - 57px)"}}, "children": [{"tagName": "Map", "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}, "children": [{"tagName": "View", "attributes": {"options": {"projection": "EPSG:3857", "center": [-100, 40], "zoom": 3.5}}, "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}}, {"tagName": "GroupLayer", "attributes": {"options": {"title": "Basemap", "fold": "close"}}, "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}, "children": [{"tagName": "TileLayer", "attributes": {"options": {"visible": false, "type": "base", "title": "None"}}, "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}, "children": [{"tagName": "XYZSource", "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}}]}, {"tagName": "TileLayer", "attributes": {"options": {"visible": true, "type": "base", "title": "OpenStreetMap"}}, "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}, "children": [{"tagName": "OSMSource", "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}}]}, {"tagName": "TileLayer", "attributes": {"options": {"visible": false, "type": "base", "title": "XYZ"}}, "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}, "children": [{"tagName": "XYZSource", "attributes": {"options": {"url": "https://maps.wikimedia.org/osm-intl/{z}/{x}/{y}.png"}}, "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}}]}, {"tagName": "TileLayer", "attributes": {"options": {"visible": false, "type": "base", "title": "NatGeo World Map"}}, "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}, "children": [{"tagName": "XYZSource", "attributes": {"options": {"url": "https://server.arcgisonline.com/ArcGIS/rest/services/NatGeo_World_Map/MapServer/tile/{z}/{y}/{x}"}}, "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}}]}, {"tagName": "TileLayer", "attributes": {"options": {"visible": false, "type": "base", "title": "USA Topo Maps"}}, "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}, "children": [{"tagName": "XYZSource", "attributes": {"options": {"url": "https://server.arcgisonline.com/ArcGIS/rest/services/USA_Topo_Maps/MapServer/tile/{z}/{y}/{x}"}}, "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}}]}, {"tagName": "TileLayer", "attributes": {"options": {"visible": false, "type": "base", "title": "World Imagery"}}, "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}, "children": [{"tagName": "XYZSource", "attributes": {"options": {"url": "https://server.arcgisonline.com/ArcGIS/rest/services/World_Imagery/MapServer/tile/{z}/{y}/{x}"}}, "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}}]}, {"tagName": "TileLayer", "attributes": {"options": {"visible": false, "type": "base", "title": "World Physical Map"}}, "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}, "children": [{"tagName": "XYZSource", "attributes": {"options": {"url": "https://server.arcgisonline.com/ArcGIS/rest/services/World_Physical_Map/MapServer/tile/{z}/{y}/{x}"}}, "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}}]}, {"tagName": "TileLayer", "attributes": {"options": {"visible": false, "type": "base", "title": "World Shaded Relief"}}, "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}, "children": [{"tagName": "XYZSource", "attributes": {"options": {"url": "https://server.arcgisonline.com/ArcGIS/rest/services/World_Shaded_Relief/MapServer/tile/{z}/{y}/{x}"}}, "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}}]}, {"tagName": "TileLayer", "attributes": {"options": {"visible": false, "type": "base", "title": "World Street Map"}}, "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}, "children": [{"tagName": "XYZSource", "attributes": {"options": {"url": "https://server.arcgisonline.com/ArcGIS/rest/services/World_Street_Map/MapServer/tile/{z}/{y}/{x}"}}, "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}}]}, {"tagName": "TileLayer", "attributes": {"options": {"visible": false, "type": "base", "title": "World Terrain Base"}}, "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}, "children": [{"tagName": "XYZSource", "attributes": {"options": {"url": "https://server.arcgisonline.com/ArcGIS/rest/services/World_Terrain_Base/MapServer/tile/{z}/{y}/{x}"}}, "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}}]}, {"tagName": "TileLayer", "attributes": {"options": {"visible": false, "type": "base", "title": "World Topo Map"}}, "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}, "children": [{"tagName": "XYZSource", "attributes": {"options": {"url": "https://server.arcgisonline.com/ArcGIS/rest/services/World_Topo_Map/MapServer/tile/{z}/{y}/{x}"}}, "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}}]}, {"tagName": "TileLayer", "attributes": {"options": {"visible": false, "type": "base", "title": "CartoDB (light_all)"}}, "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}, "children": [{"tagName": "XYZSource", "attributes": {"options": {"url": "http://{1-4}.basemaps.cartocdn.com/light_all/{z}/{x}/{y}.png"}}, "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}}]}, {"tagName": "TileLayer", "attributes": {"options": {"visible": false, "type": "base", "title": "CartoDB (light_nolabels)"}}, "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}, "children": [{"tagName": "XYZSource", "attributes": {"options": {"url": "http://{1-4}.basemaps.cartocdn.com/light_nolabels/{z}/{x}/{y}.png"}}, "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}}]}, {"tagName": "TileLayer", "attributes": {"options": {"visible": false, "type": "base", "title": "CartoDB (dark_all)"}}, "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}, "children": [{"tagName": "XYZSource", "attributes": {"options": {"url": "http://{1-4}.basemaps.cartocdn.com/dark_all/{z}/{x}/{y}.png"}}, "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}}]}, {"tagName": "TileLayer", "attributes": {"options": {"visible": false, "type": "base", "title": "CartoDB (dark_nolabels)"}}, "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}, "children": [{"tagName": "XYZSource", "attributes": {"options": {"url": "http://{1-4}.basemaps.cartocdn.com/dark_nolabels/{z}/{x}/{y}.png"}}, "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}}]}]}, {"tagName": "GroupLayer", "attributes": {"options": {"title": "Overlays", "fold": "open"}}, "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}, "children": [{"tagName": "ImageLayer", "attributes": {"options": {"title": "GEOGLOWS Streamflow Service"}}, "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}, "children": [{"tagName": "ImageArcGISRestSource", "attributes": {"options": {"url": "https://livefeeds3.arcgis.com/arcgis/rest/services/GEOGLOWS/GlobalWaterModel_Medium/MapServer"}}, "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}}]}]}, {"tagName": "ScaleLineControl", "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}}, {"tagName": "LayerPanel", "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}}]}, {"tagName": "div", "attributes": {"role": "dialog", "aria-modal": "true", "className": "offcanvas offcanvas-end show", "tabIndex": "-1", "style": {"visibility": "visible", "width": "50vw"}}, "children": [{"tagName": "div", "attributes": {"className": "offcanvas-header"}, "children": [{"tagName": "div", "attributes": {"className": "offcanvas-title h5"}, "children": ["Forecast"]}, {"tagName": "button", "attributes": {"type": "button", "className": "btn-close", "aria-label": "true"}, "eventHandlers": {"onClick": null}}]}, {"tagName": "div", "attributes": {"className": "offcanvas-body"}, "children": [{"tagName": "Container", "children": [{"tagName": "Row", "children": [{"tagName": "Col", "children": [{"tagName": "h2", "children": ["Streamflow @ Test 123"]}, {"tagName": "LineChart", "attributes": {"width": 700, "height": 500, "data": [{"x": 1, "y": 100}, {"x": 2, "y": 200}]}, "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}, "children": [{"tagName": "CartesianGrid", "attributes": {"strokeDasharray": "3 3"}, "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}}, {"tagName": "XAxis", "attributes": {"label": "Date"}, "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}}, {"tagName": "YAxis", "attributes": {"label": {"value": "Streamflow", "angle": -90, "position": "insideLeft"}}, "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}}, {"tagName": "Tooltip", "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}}, {"tagName": "Line", "attributes": {"type": "monotone", "dataKey": "y"}, "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}}]}], "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}}], "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}}], "importSource": {"source": "test_page_2.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}}]}]}]}